### ISSUE TITLE
VNET Integrated Logic App - Unable to create a Logic App SQL (V2) connection

### ISSUE THREAD

**Issue by slydewd** [Azure/logicapps#1089]
### Describe the Bug

I am trying to setup a VNET integrated Logic App (Standard) that will connect to our onpremises SQL database using the SQL (V2) connectors. We are unable to create a Logic App SQL (V2) connection using the VNET integrated Logic App. 

### Plan Type

Standard

### Steps to Reproduce the Bug or Issue

1. Create Standard Logic App that's VNET integrated.
2. Open Azure and onprem firewall from Logic App subnet to onprem SQL Server over port 1433
4. Create a workflow starting with an HTTP request trigger to manually be able to start the workflow
5. Add an Action to the workflow > Search for "SQL" > Choose any of the SQL connectors with a (V2) in the name
6. Try creating a connection in the SQL (V2) action using SQL Server authentication.
7. You'll encounter a couple of errors depending on the server name you add to the connection. You can see the different errors in the screenshots below.

### Workflow JSON

```json
{
    "definition": {
        "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "actions": {},
        "contentVersion": "1.0.0.0",
        "outputs": {},
        "triggers": {
            "manual": {
                "kind": "Http",
                "type": "Request"
            }
        }
    },
    "kind": "Stateful"
}
```

### Screenshots or Videos

**V2 connector I am not able to save because the connection isn't being created**
![2024-06-19 10_58_35-vk-sqlv2 - Microsoft Azure — Mozilla Firefox](https://github.com/Azure/logicapps/assets/96139038/54833ed2-88ce-403c-8a6b-5aca0d179d39)
**Error when using only IP in server name**
![2024-06-19 10_01_33-vk-sqlv2 - Microsoft Azure — Mozilla Firefox](https://github.com/Azure/logicapps/assets/96139038/7c1b79e0-7423-4e1f-9a52-73296aaa5906)
**Error when using IP[:PORT] in server name (IDs are hidden)**
![2024-06-19 10_03_21-vk-sqlv2 - Microsoft Azure — Mozilla Firefox](https://github.com/Azure/logicapps/assets/96139038/17f07bf5-0ff0-413a-b1b4-bbd000d9eaaa)
**V1 connector showing successful run**
![2024-06-19 10_47_01-vk-sqlv1 - Microsoft Azure — Mozilla Firefox](https://github.com/Azure/logicapps/assets/96139038/87b7cb17-fc29-4a94-bfcb-5e8f402c41d7)

### Additional context

The workflow JSON isn't showing the SQL (V2) action because I am unable to save the workflow since I am not able to create a connection.

**Architecture**
* Logic App > NSG > Azure Firewall > Onprem Firewall > SQL Server

**Authentication**
* SQL Server authentication without onprem data gateway.

**Tried/Done**
* Opened Azure and onprem firewall. It's opened from Logic App subnet to onprem SQL server over port 1433.
* Azure firewall logs isn't showing any traffic to the destination IP using V2.
* Tried using host name and IP, also trying with and without domain and port.

**What works**
* Setting it up using the V1 connector instead of V2 works without any issues, but we cannot use this as the V1 connector will retire next year...

**Comment by slydewd** [comment]
I contacted Microsoft support and they solved it.

As I explained in my post, the V1 was always working without any gateway. What they explained was that the connector I used wasn't the V1 and a totally different connector than the one with V2 in its name. The one that has V2 is an Azure connector and the other I was using is an in-app connector (Not V1!). The Azure connectors needs to use a gateway to access onprem as it will default to using public IPs even though the Logic App is VNET integrated. The in-app connector is able to use private IPs without any gateway if the app is VNET integrated.
